internal/monitors/collectd/df/df.go

```diff
@@ -49,7 +49,36 @@ type Monitor struct {
 	collectd.MonitorCore
 }
 
+// GetExtraMetrics returns additional metrics to allow through.
+func (c *Config) GetExtraMetrics() []string {
+	var extraMetrics []string
+	if c.ReportInodes {
+		extraMetrics = append(extraMetrics, groupMetricsMap[groupReportInodes]...)
+	}
+	if c.ValuesPercentage {
+		extraMetrics = append(extraMetrics, groupMetricsMap[groupValuesPercentage]...)
+	}
+	if c.ReportInodes && c.ValuesPercentage {
+		extraMetrics = append(extraMetrics, []string{percentInodesFree, percentInodesReserved, percentInodesUsed}...)
+	}
+	return extraMetrics
+}
+
 // Configure configures and runs the plugin in collectd
-func (m *Monitor) Configure(conf *Config) error {
-	return m.SetConfigurationAndRun(conf)
+func (m *Monitor) Configure(config *Config) error {
+	// conf is a config shallow copy that will be mutated and used to configure monitor
+	conf := *config
+	// Setting group flags in conf for enable extra metrics
+	for _, metric := range conf.EnabledMetrics {
+		switch metricSet[metric].Group {
+		case groupReportInodes:
+			conf.ReportInodes = true
+		case groupValuesPercentage:
+			conf.ValuesPercentage = true
+		default:
+			conf.ReportInodes = true
```

Comment:
`df_complex.free` and others don't have groups so they'd end up enabling both config flags I think.